### ISSUE TITLE
The BBA validate link in the concierge DM is broken

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#34284]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.24-0
**Reproducible in staging?:** y
**Reproducible in production?:** can't check
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @anmurali 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1704820280176009

## Action Performed:
1. Create a workspace and add a fully verified business bank account 
2. Go to the concierge DM, where you are asked to validate the account you added by entering the amounts of the 3 transactions sent to your bank account.
3. Click on the link

## Expected Result: 
Clicking on the link should take you to your workspace settings - bank account
## Actual Result: 
Clicking on the link shows a Hmm... it's not here modal.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![Screen Shot 2024-01-10 at 1 31 59 PM](https://github.com/Expensify/App/assets/38435837/fb7b54b5-6934-4170-82fc-0b3ca148b5b8)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by ikevin127** [comment]
This looks back-end related since that's where the Concierge messages come from.

Allow me to explain:
- from the screenshot we can notice the URL being `https://staging.new.expensify.com/bank-account/requestor` which I assume Concierge supplied to the user in a message
- most likely the reason why the user is seeing the notAuthorized screen is due to the user having multiple policies and the URL is missing the policyID param ex. `?policyID=CDAEX611488AD952` at the end of `.../requestor`
- this results in the first policy being used by default which makes the `!PolicyUtils.isPolicyAdmin(policy)` true in the code block below which in turn displays the notAuthorized screen as the if passes
https://github.com/Expensify/App/blob/8460043e6251a8a8f5dc35f47381589d5baeee7d/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L400

That being said, I think the bank account related Concierge URL's that come from the back-end should contain the policyID param like they do in other real cases like the one in the image below.
<img width="304" alt="Screenshot 2024-01-11 at 04 07 38" src="https://github.com/Expensify/App/assets/56457735/9561f6ed-8933-4428-86df-3f3b42a8c47e">

**Comment by sophiepintoraetz** [comment]
@aimane-chnaif - mind confirming that this is internal? I can see if we want to fix this.

**Comment by marcaaron** [comment]
Looks like something we should definitely fix. I don't quite have the bandwidth to take this on right now as a `Daily`.

It looks like the description needs to be updated.

>Go to the concierge DM, where you are asked to validate the account you added by entering the amounts of the 3 transactions sent to your bank account.

In this case, it looks like they are being brought to the requestor step because there is some issue with the KYC information. Still, they should not see this "not found" page.

Agree with the assessment from @ikevin127 that if we want this to work then we'd need to have a policyID in the route params and that it makes sense that the policy would be specified in the message from Concierge.

IIRC this flow was originally designed to support just a single workspace and not multiple workspaces. 

Maybe @MariaHCD or @nkuoch would more easily see what the issue is here. if not, I can try to take a deeper look in the coming weeks.

**Comment by marcaaron** [comment]
Looks like [here](https://github.com/Expensify/Web-Expensify/blob/8e78af5678c49712fb9208ecd9b971544c67d2a9/lib/BankAccount/VerificationUtils.php#L295-L320) is where the link is built. But confused about how that was working before without a `policyID` 🤔

**Comment by ikevin127** [comment]
> the first policy being used by default

This is why it does work without `policyID` in the link but only when the policies object contains 1 policy and its the expected one where the current user isAdmin, then it navigates to that one by default.

When I reproduced the issue, the account I used had 3 policies, and the first one was one in which `isAdmin` was `false`.

That one was navigated to which displayed the notAuthorized screen in RHN.

That's why some accounts will not see the notAuthorized screen and some will and that's why we should always pass policyID as param in the links from Concierge because of this edge case.

**Comment by nkuoch** [comment]
> IIRC this flow was originally designed to support just a single workspace and not multiple workspaces.

Who is supposed to have worked on supporting multiple workspaces? Because in the code, I still don't see it supported...

**Comment by marcaaron** [comment]
Thanks! And yeah, this feels more like a "new feature" then. 

It feels like an unexpected side effect of having multiple workspaces. The flow was created for the free plan and optimized around that. Maybe we should give some more thought here to how to prioritize this issue.

@anmurali should this perhaps be moved into Wave 8 or 9?

**Comment by marcaaron** [comment]
>I actually don't quite understand why we originally created a EXPENSIFY_FREE_PLAN_BANK_ACCOUNT NVP instead of just putting it in the policy achAccount as we did in oldDot? Or was it meant to be one single bank account to be the same across all workspaces?

I think you could have persuaded us to do it that way 😄  It sounds better in retrospect and in the context of the current roadmap.

At the time we were just optimizing for a user that:

- Multiple workspaces and multiple bank account support was not a priority yet
- Would have a single workspace + VBBA
- Sign up for the free plan
- Get Expensify Cards
- Never really does anything else

Probably we could have still done all that without the NVP - though it gave us a quick way to get the correct free plan bank account for the workspace they set up in NewDot vs some other bank account.

**Comment by marcaaron** [comment]
@nkuoch  Will the linked PRs will fix this? Or do we still have to add something in Web-E? Lemme know if you want me to look into anything,

**Comment by nkuoch** [comment]
It will be fixed by https://github.com/Expensify/Web-Expensify/pull/40751, which will add the right url (https://staging.new.expensify.com/bank-account/validate/?policyID=XXX) in this message
<img width="1349" alt="image" src="https://github.com/Expensify/App/assets/2463975/f78fbf20-5601-40cd-9589-72580e6b01f5">

**Comment by nkuoch** [comment]
Sorry, it will be really fixed only once https://github.com/Expensify/Expensify/issues/299712 is finished.

**Comment by laurenreidexpensify** [comment]
Hold on https://github.com/Expensify/Expensify/issues/299712

**Comment by laurenreidexpensify** [comment]
Still Holding on https://github.com/Expensify/Expensify/issues/299712

**Comment by laurenreidexpensify** [comment]
Still Holding on https://github.com/Expensify/Expensify/issues/299712

**Comment by laurenreidexpensify** [comment]
https://github.com/Expensify/Expensify/issues/299712 is now deployed to prod - @nkuoch @marcaaron we can take this off hold now right?
